### PR TITLE
TBDGen: avoid emitting symbols with OBJC_CLASS and OBJC_METACLASS prefixes

### PR DESCRIPTION
**Pull request by nkcsgexi**
rdar://60057489

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/TBDGen/TBDGen.cpp [hunk 1]
```diff
@@ -56,16 +56,28 @@ using namespace llvm::yaml;
...
+static constexpr StringLiteral ObjC2MetaClassNamePrefix = "_OBJC_METACLASS_$_";
+
 static bool isGlobalOrStaticVar(VarDecl *VD) {
   return VD->isStatic() || VD->getDeclContext()->isModuleScopeContext();
 }
 
+// If a symbol is implied, we don't need to emit it explictly into the tbd file.
+// e.g. When a symbol is in the `objc-classes` section in a tbd file, a additional
+// symbol with `_OBJC_CLASS_$_` is implied.
+static bool isSymbolImplied(StringRef name) {
```
Comment (slavapestov) [inline lib/TBDGen/TBDGen.cpp]: Instead, please just comment out these two calls elsewhere in this file:

```
addSymbol(LinkEntity::forObjCClass(CD));
addSymbol(LinkEntity::forObjCMetaclass(CD));
```

Comment (nkcsgexi) [reply lib/TBDGen/TBDGen.cpp]: We do need them in the symbol sets for verifying against irgen.

File: test/TBD/implied_objc_symbols.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: VENDOR=apple
+// RUN: %empty-directory(%t)
+
+// RUN: echo "import Foundation" > %t/main.swift
+// RUN: echo "@objc public class ObjCOnly {}" >> %t/main.swift
```
Comment (cyndyishida) [inline test/TBD/implied_objc_symbols.swift]: can you add a check for existence of the symbol name too, not just that the prefix doesn't appear?

Comment (nkcsgexi) [reply test/TBD/implied_objc_symbols.swift]: Done

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: test/TBD/implied_objc_symbols.swift (hunk 2 above)
Comment (cyndyishida) [reply test/TBD/implied_objc_symbols.swift]: thanks!

**Review by cyndyishida — APPROVED** [review]

**Pull request merged**